Mathlib/Algebra/Order/Group/Cyclic.lean

```diff
@@ -27,6 +28,30 @@ variable {G : Type*} [CommGroup G] [LinearOrder G] [IsOrderedMonoid G]
 
 namespace Subgroup
 
+@[to_additive]
```

Comment:
I would guess a proof can be done with way less hypotheses, and the result should be useful: `Group G`only, and no linear order, it is just needed that the elements don't have finite order (one of them suffices).

Reply:
On the other hand you need to make sure it's easy to apply. Maybe `[IsMulTorsionFree G]` is the right hypothesis and then one should check that there's an instance `[Group G] [LinearOrder G] [IsOrderedMonoid G] -> IsMulTorsionFree G`

Reply:
Using `\not IsOfFinOrder x` as an hypothesis seems to work well. Under ` [LinearOrder G] [IsOrderedMonoid G]`, there is indeed an instance `IsMulTorsionFree G`, and then one can use `not_isOfFinOrder_of_isMulTorsionFree`.